docs/docs/configuration/overview.md

```diff
@@ -190,8 +190,9 @@ Provider specific options can be found on their respective subpages.
 | ------------------------------------------------------------------------- | -------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------- |
 | flag: `--allow-query-semicolons`<br/>toml: `allow_query_semicolons`       | bool           | allow the use of semicolons in query args ([required for some legacy applications](https://github.com/golang/go/issues/25192))                                                                                                                                                                                                                                                                                                                                                                                        | `false`     |
-| flag: `--api-route`<br/>toml: `api_routes`                                | string \| list | return HTTP 401 instead of redirecting to authentication server if token is not valid. Format: path_regex                                                                                                                                                                                                                                                                                                                                                                                                             |             |
+| flag: `--api-route`<br/>toml: `api_routes`                                | string \| list | Requests to these paths must already be authenticated with a cookie, or a JWT if `--skip-jwt-bearer-tokens` is set. No redirect to login will be done. Return 401 if not. Format: path_regex                                                                                                                                                                                                                                                                                                                                                                                                             |             |
```

Comment:
This old description actually sounds like what I'm implementing in this PR, but the way I read it [here](https://github.com/oauth2-proxy/oauth2-proxy/blob/master/oauthproxy.go#L1019) and [here](https://github.com/oauth2-proxy/oauth2-proxy/blob/master/oauthproxy.go#L1113), this option can't distinguish between an invalid token and no token at all -- `ErrNeedsLogin` is returned in either case. Some better tests are called for here I believe, including how this interacts with my new option. If you think that's necessary for this PR, please let me know.